samples/bluetooth/fast_pair/locator_tag/src/fp_adv.c

```diff
@@ -527,12 +556,48 @@ int app_fp_adv_init(void)
 		return err;
 	}
 
+	/* Start the advertising if any module already requested it. */
+	fp_adv_request_start();
+
+	is_enabled = true;
+
+	LOG_INF("Fast Pair: fp_adv module enabled");
+
 	return 0;
 }
 
-bool app_fp_adv_is_init(void)
+int app_fp_adv_disable(void)
 {
-	return (fp_adv_set != NULL);
+	int err;
+
+	__ASSERT_NO_MSG(is_initialized);
+
+	if (!is_enabled) {
+		LOG_ERR("Fast Pair: fp_adv module already disabled");
+		return 0;
+	}
+
+	is_enabled = false;
+
+	/* Suspend the requested advertising until the fp_adv module reinitializes. */
+	fp_adv_mode_set(APP_FP_ADV_MODE_OFF);
+
+	app_fp_adv_rpa_rotation_suspend(false);
```

Comment:
I wonder if we should actually clear it (inconsistent with handling fp_adv requests which are not automatically cleared)

Reply:
Discussed during standup. We will move it to factory reset prepare callback of main.